src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/CacheNode.java

```diff
@@ -0,0 +1,75 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * The base class for all cache nodes
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public abstract class CacheNode {
+
+    private final String key;
+
+    private final Collection<String> parents;
```

Comment:
It makes building a graph tricky because you don't have all cache nodes to refer to...